test/unit/test_sign.py

```diff
@@ -116,3 +122,27 @@ def test_identity_proof_claim_lookup(id_config, monkeypatch):
     assert expected_entry.integrated_time == actual_entry.integrated_time
     assert expected_entry.log_id.key_id == bytes.fromhex(actual_entry.log_id)
     assert expected_entry.log_index == actual_entry.log_index
+
+
+@pytest.mark.online
+@pytest.mark.ambient_oidc
+def test_sign_prehashed(staging):
+    sign_ctx, verifier, identity = staging
+
+    sign_ctx: SigningContext = sign_ctx()
+    verifier: Verifier = verifier()
+
+    input_ = io.BytesIO(secrets.token_bytes(32))
+    hashed = Hashed(digest=sha256_streaming(input_), algorithm=HashAlgorithm.SHA2_256)
+
+    with sign_ctx.signer(identity) as signer:
+        bundle = signer.sign(hashed)
+
+    assert bundle.message_signature.message_digest.algorithm == hashed.algorithm
+    assert bundle.message_signature.message_digest.digest == hashed.digest
+
+    input_.seek(0)
+    materials = VerificationMaterials.from_bundle(
+        input_=input_, bundle=bundle, offline=False
+    )
+    verifier.verify(materials=materials, policy=UnsafeNoOp())
```

Comment:
I've added this to confirm that the round-trip works as expected: we sign over a `Hashed(...)` and then verify over the pre-image.